### PR TITLE
fix: improve k3s upgrade, more cli opts, updates k3s env

### PR DESCRIPTION
**Pull request by bohdand-weka**

### CODE REVIEW

File: internal/cli/local/setup/flags/flags.go [hunk 1]
```diff
@@ -29,6 +29,11 @@ type Flags struct {
 		RemoteVersion  string
 	}
 
+	Iface string
```
Comment (bohdand-weka) [inline internal/cli/local/setup/flags/flags.go:32]: those flags moved to common

File: internal/cli/local/setup/run.go [hunk 2]
```diff
@@ -55,7 +55,7 @@ func runSetup(cmd *cobra.Command, args []string) (err error) {
 		}
 	}()
 
-	err = k3s.Install(cmd.Context(), k3s.InstallConfig{
+	err = k3s.Install(cmd.Context(), k3s.Config{
```
Comment (bohdand-weka) [inline internal/cli/local/setup/run.go:58]: now install config and upgrade config are the same

File: internal/local/k3s/install.go [hunk 3]
```diff
@@ -171,84 +128,11 @@ func copyAirgapImages() bundle.TarCallback {
...
-			}
-
-			if err = cmd.Wait(); err != nil {
-				return fmt.Errorf("install.sh: %w", errors.Join(err, ctx.Err()))
-			}
-
-			logger.Info().Msg("Install completed")
-
-			return nil
+			return k3sInstall(ctx, c, fi, r)
```
Comment (bohdand-weka) [inline internal/local/k3s/install.go:135]: moved to shared func

File: internal/local/k3s/k3s.go [hunk 4]
```diff
@@ -19,15 +20,51 @@ import (
...
 	"github.com/weka/gohomecli/internal/local/bundle"
+	config_v1 "github.com/weka/gohomecli/internal/local/config/v1"
 	"github.com/weka/gohomecli/internal/utils"
 )
 
 const (
 	k3sInstallPath = "/usr/local/bin"
 )
 
+type Config struct {
```
Comment (bohdand-weka) [inline internal/local/k3s/k3s.go:31]: moved to k3s file from install

File: internal/local/web/api/upgrade.go [hunk 5]
```diff
@@ -20,7 +20,7 @@ func upgrade(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	err := k3s.Upgrade(r.Context(), k3s.UpgradeConfig{})
+	err := k3s.Upgrade(r.Context(), k3s.Config{})
```
Comment (bohdand-weka) [inline internal/local/web/api/upgrade.go:23]: web is still not implemented

**Review by gonenidan — APPROVED** [review]

**Pull request merged**